File: src/main/java/employeetech/com/employee/manager/model/Employee.java
package employeetech.com.employee.manager.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Employee implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false, updatable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id")
    private Department department;

    private String email;
    private String title;
    private String phone;
    private String name;
    private String imageUrl;

    @Column(nullable = false, updatable = false)
    private String employeeCode;

    @Override
    public String toString() {
        return "Employee(id=" + id + ", name=" + name + ", email=" + email + 
               ", title=" + title + ", phone=" + phone + ", imageUrl=" + imageUrl + 
               ", employeeCode=" + employeeCode + ")";
    }
}

File: src/main/kotlin/employeetech/com/employee/manager/model/Employee.kt
package employeetech.com.employee.manager.model

import jakarta.persistence.*
import java.io.Serializable

@Entity
data class Employee(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false, updatable = false)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id")
    val department: Department? = null,

    val email: String? = null,
    val title: String? = null,
    val phone: String? = null,
    val name: String? = null,
    val imageUrl: String? = null,

    @Column(nullable = false, updatable = false)
    val employeeCode: String? = null
) : Serializable {

    override fun toString(): String {
        return "Employee(id=$id, name=$name, email=$email, title=$title, phone=$phone, imageUrl=$imageUrl, employeeCode=$employeeCode)"
    }
}
